src/main/java/com/google/devtools/build/lib/analysis/starlark/Args.java

```diff
@@ -573,8 +601,10 @@ private MutableArgs(@Nullable Mutability mutability, StarlarkSemantics starlarkS
     }
 
     @Override
-    public CommandLine build() {
-      return commandLine.build(flagPerLine);
+    public CommandLine build(InterruptibleRepoMappingSupplier mainRepoMappingSupplier)
```

Comment:
Take the supplier up front in `Args.newArgs()` which only has one caller. That way `build()` can be kept parameterless.

Reply:
This is what I did at first, but then I got worried about what could happen if someone creates an `Args` in one rule context (say, command line arguments shared across targets via a toolchain) and then registers it for an action in another context. The supplier would then try to access the analysis environment of the first context - can this work?

I implemented the suggestion for now, but can always revert to the previous state.

Reply:
Good thinking. I didn't think of that possibility. It does in fact work currently if you try to share an `Args` through a provider, so long as the `Args` is not mutated outside of the rule context that created it. This is because `build()` doesn't do a mutability check.

It would be great if we had time to invest in adding that mutability check to `build()` to prohibit this, but I won't make you do that. Feel free to revert to the previous methodology, but consider a helper method for the boilerplate of making that supplier out of a method reference.

Reply:
Is it clear that we want to prevent `Args` sharing via a provider? I could see that saving memory by e.g. sharing flags via a toolchain, so folks may be relying on this being possible.

Reply:
It's not clear, no. I thought it was odd that it is possible to share what amounts to an immutable builder, but that's my opinion, and shouldn't distract from your effort.

Reply:
I reverted to the previous approach with a helper method, PTAL.